### PR TITLE
Support contract mapping

### PR DESCRIPTION
**Pull request by nolag**

### CODE REVIEW

File: pkg/codec/modifier_codec_test.go [hunk 1]
```diff
@@ -140,7 +140,7 @@ func TestModifierCodec(t *testing.T) {
 	})
 
 	var actual any
-	t.Run("CreateType returns modified type", func(t *testing.T) {
+	t.Run("CreateContractType returns modified type", func(t *testing.T) {
```
Comment (ilija42) [inline pkg/codec/modifier_codec_test.go:143]: nit, why was this renamed, it stil calls CreateType

**Review by ilija42 — APPROVED** [review]
approved with a small question/nit

File: pkg/codec/modifier_codec_test.go (hunk 1 above)
Comment (nolag) [reply pkg/codec/modifier_codec_test.go:143]: For tests in common, it does.  We need the Contract and the TypeName  for the real implementation when it's chain reader.
